### PR TITLE
Ymd40v2new layouts and fixed per-switch backlight

### PR DESCRIPTION
**Pull request by trenthoh**
Backlight wasn't working for the default.
Multimeter-tested thanks to advice from Xelus and JayBau. Pin changed from B6 to B7.

Added ymd40v2_2x2u ymd40v2_1x2uR ymd40v2_1x2uL ymd40v2_1x2uC following planck convention

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/ymdk/ymd40/v2/v2.h [hunk 1]
```diff
@@ -29,3 +29,51 @@
     { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B   }, \
     { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B   }, \
 }
+
+#define LAYOUT_ymd40v2_2x2u( \
```
Comment (fauxpark) [inline keyboards/ymdk/ymd40/v2/v2.h:33]: These should be named `LAYOUT_ortho_4x12_*` to match the existing layout.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ymdk/ymd40/v2/v2.h (hunk 1 above)
Comment (trenthoh) [reply keyboards/ymdk/ymd40/v2/v2.h:33]: Hi Ryan,

I was just inferring from using the planck as a baseline.
LAYOUT_planck_2x2u
LAYOUT_ortho_4x12

Should I proceed with changing the names as per your suggestion?

Comment (fauxpark) [reply keyboards/ymdk/ymd40/v2/v2.h:33]: Yeah, those Planck layout names are not ideal, we prefer using more generic names so they can be reused for any other keyboard with the same layout - see the Layouts documentation: https://docs.qmk.fm/#/feature_layouts

**Review by fauxpark — APPROVED** [review]

File: keyboards/ymdk/ymd40/v2/v2.h (hunk 1 above)
Comment (trenthoh) [reply keyboards/ymdk/ymd40/v2/v2.h]: Thanks for the advice.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**